src/clarity-angular/layout/tabs/tabs.ts

```diff
@@ -3,35 +3,84 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {Component} from "@angular/core";
+import {
+    Component, ContentChildren, QueryList, ViewChild, ViewContainerRef
+} from "@angular/core";
 import {IfActiveService} from "../../utils/conditional/if-active.service";
 import {Tab} from "./tab";
 import {TabsService} from "./tabs-service";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+import {TabLinkDirective} from "./tab-link.directive";
 
-// TODO: find a better way to separately render the tab contents (i.e. not inside the ul)
 @Component({
     selector: "clr-tabs",
-    template: `
+    template: `        
         <ul class="nav" role="tablist">
-            <ng-content select="clr-tab"></ng-content>
+            <ng-container #inlineTabLinks></ng-container>
+            <!--this is the part that goes into the overflow-->
+            <ng-container *ngIf="tabsService.overflowTabs.length > 0">
+                <div class="tabs-overflow bottom-right" [class.open]="ifOpenService.open" 
+                     (click)="toggleOverflow($event)">
+                    <li role="presentation" class="nav-item">
+                        <button class="btn btn-link nav-link dropdown-toggle" [class.active]="activeTabInOverflow">
+                            <clr-icon shape="ellipsis-horizontal" [class.is-info]="ifOpenService.open"></clr-icon>
+                        </button>
+                    </li>
+                    <clr-tab-overflow-content>
+                        <ng-container #overflowTabLinks></ng-container>
+                    </clr-tab-overflow-content>
+                </div>
+            </ng-container>
         </ul>
+        <!--this is the tab content, rendered outside the ul element above-->
+        <!--TODO: find a better way to achieve this-->
         <ng-container *ngFor="let tab of tabsService.children">
             <ng-template *ngIf="tab.tabContent" [ngTemplateOutlet]="tab.tabContent.templateRef"></ng-template>
         </ng-container>
```

Comment:
We can get rid of that now!

Since you are now "extracting" the links rather than the content, you can project the `<clr-tab>` components themselves directly with `<ng-content>`. You don't even need a selector on it.